sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/package.scala

```diff
@@ -159,6 +159,19 @@ package object util {
   def toCommentSafeString(str: String): String =
     str.replace("*/", "\\*\\/").replace("\\u", "\\\\u")
 
+  /**
+   * Returns the string representation of this expression with origin that is safe to be put in
+   * code comments of generated code.
+   */
+  def toCommentSafeString(expr: Expression): String = {
+    val str = if (expr.origin.callSite.isDefined && !expr.isInstanceOf[BoundReference]) {
+      expr.toString + " @ " + expr.origin.callSite.get
+    } else {
+      expr.toString
+    }
+    str.replace("*/", "\\*\\/").replace("\\u", "\\\\u")
```

Comment:
I think it's better not to overload `toCommentSafeString`. Instead of that, how about add another method for making origin information string like `toOriginString` to `Expression` and call the method like `toCommentSafeString(expr.toOriginString())`?
It's just one idea.